emr-user-role-mapper-application/src/test/com/amazon/aws/emr/integration/defaultmapper/DefaultMappingProviderImplIntegrationTest.java

```diff
@@ -0,0 +1,197 @@
+    jerseyServlet.setInitOrder(0);
+
+    // Tells the Jersey Servlet which REST service/class to load.
+    jerseyServlet.setInitParameter("jersey.config.server.provider.packages", "com.amazon.emr.api");
+    jerseyServlet
+        .setInitParameter("javax.ws.rs.Application",
+            DefaultProviderImplIntegrationApplication.class.getName());
+
+    log.debug("Starting the Jetty server");
+    jettyServer.start();
+    // AWS Role creation is eventually consistent.
+    // TODO - Fine tune this.
+    Thread.sleep(10 * 1000);
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    if (jettyServer != null) {
+      jettyServer.stop();
+      jettyServer.join();
+      jettyServer.destroy();
+      jettyServer = null;
+    }
+    // Role deletion requires no policies to be attached.
+    if (testRoleArn != null && testPolicyArn != null) {
+      IAMUtils.detachPolicyFromRole(testRoleName, testPolicyArn);
+    }
+    if (testRoleArn != null) {
+      IAMUtils.deleteRole(testRoleName);
+    }
+    if (testPolicyArn != null) {
+      IAMUtils.deletePolicy(testPolicyArn);
+    }
+  }
+
+  @Before
+  public void beforeMethod() {
+    // Skip tests if the underlying OS is not supported.
+    Assume.assumeFalse(!isOsSupported());
+  }
+
+  @Test
+  public void list_roles_api() throws Exception {
+    HttpUriRequest request =
+        new HttpGet(LOCALHOST_SERVER + ":" + TEST_PORT + IMDS_CREDENTIALS_URI);
+    HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);
+    assertThat(httpResponse.getStatusLine().getStatusCode(), is(HttpStatus.SC_OK));
+    String responseString = new BasicResponseHandler().handleResponse(httpResponse);
+    assertThat(responseString, is(testRoleName));
+  }
+
+  @Test
+  public void credentials_api() throws Exception {
+    HttpUriRequest request =
+        new HttpGet(LOCALHOST_SERVER + ":" + TEST_PORT + IMDS_CREDENTIALS_URI + testRoleName);
+    HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);
+    assertThat(httpResponse.getStatusLine().getStatusCode(), is(HttpStatus.SC_OK));
+    String responseString = new BasicResponseHandler().handleResponse(httpResponse);
+    EC2MetadataUtils.IAMSecurityCredential credentials = GSON.fromJson(responseString,
+        EC2MetadataUtils.IAMSecurityCredential.class);
+    BasicSessionCredentials sessionCredentials = new BasicSessionCredentials(
+        credentials.accessKeyId,
+        credentials.secretAccessKey,
+        credentials.token);
+
+    AmazonS3 s3 = AmazonS3ClientBuilder.standard()
+        .withCredentials(new AWSStaticCredentialsProvider(sessionCredentials))
+        .build();
+    S3Object s3Object = s3.getObject(new GetObjectRequest(TEST_CFG_BUCKET, TEST_CFG_OBJECT));
+    assertThat(S3Utils.getS3FileAsString(s3Object), is(TEST_S3_OBJECT_CONTENTS));
+  }
+
+  @Test
+  public void reload_config_no_mapping_no_credentials() throws Exception {
```

Comment:
I think this test is trying to test on the case where URM periodically syncs the mapping file from s3. Should we rename this file to be more explicit? Especially, this test does not verify anything around credential or empty mapping.

Reply:
So what the test does is 
- mapping present, check we get credentials
- Upload a new mapping file where user has no role mapping
- Check that we get back empty credentials
Renamed the test to make the intention clearer.

Reply:
Changed the name of test.